python/tank_vendor/shotgun_authentication/interactive_authentication.py

```diff
@@ -330,50 +349,59 @@ def _process_ui():
 
 # Lock the assures only one thread at a time can execute the authentication logic.
 _renew_session_internal_lock = threading.Lock()
```

Comment:
Wow, I didn't know about the range(3) trick. Awesome!
After your's  and Rob's comment about the session cache becoming a class, I started wondering if this piece of code shouldn't be a class as well. I'll definitely update that.
Thanks!